### ISSUE TITLE
Player is lagging since 0.5.9

### ISSUE THREAD

**Issue by Richard38907** [Malopieds/InnerTune#304]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/Malopieds/InnerTune/actions).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce the bug

Opening animation of the mini-player is lagging since v 0.5.9

### Expected behavior

Fluid animation

### Actual behavior

Lagging animation

### Logs

[logcat.txt](https://github.com/user-attachments/files/16603982/logcat.txt)

### InnerTune version

0.5.9

### Android version

Android 13

**Comment by Malopieds** [comment]
Could you show me how noticeable it is with a screen record? I don't see any major differences.

**Comment by Richard38907** [comment]
Sadly, screen record doesn't render the issue because the video recording makes my phone lagging too 😅😅. I use OnePlus Nord and I notice a noticeable UI performance drop compared to 0.5.8 or z-Huang's version.

**Comment by Richard38907** [comment]
By the way. "Gradient player" theme makes the player lag too since the release of this theme.

**Comment by Malopieds** [comment]
I'm using a 90Hz screen and I mean I don't really see a difference by the simple eye between my fork and the original. Plus the gradient theme has been there for more than a month and no complaint :/
What phone are you using?

**Comment by Richard38907** [comment]
OnePlus Nord with Pixel experience 13 ROM installed.

**Comment by Richard38907** [comment]
~~I tested on my OnePlus 10t ( OxygenOS 14) now and the glitch is not present. Weird🤔~~

**Comment by Richard38907** [comment]
No I understand. It's only on AMOLED mode. It lags on two phones with this mode activated.
![Screenshot_20240813-222442_InnerTune.png](https://github.com/user-attachments/assets/89bf1882-8f44-42cd-95eb-be393bbbc204)
When the player is in pure black theme, the glitch occurs.

**Comment by Malopieds** [comment]
Yeah since AMLOED dark pixel are actually turned off, it might be the cause of "lagging" (which will be in that case just the amoled pixels that are turning on)

**Comment by Richard38907** [comment]
I use a lot of apps with pure black mode and they don't lag. The Z-Huang latest version has the pure black player and it doesn't lag. The difference is really noticeable👁️.

**Comment by Malopieds** [comment]
There is only one difference from z-huang black theme, it's the moment to which I activate the dark mode, but that's what I found frustrating with their implementation, you sometime get the dark mini player when it shouldn't :/

**Comment by Richard38907** [comment]
I really would like to help you more to find the cause of this issue but I have no experience in software coding.

**Comment by d4f5409d** [comment]
I see gradient often loading a whole lot after the start of the song.. I don't know what's the cause of this issue...

**Comment by d4f5409d** [comment]
And when I swipe up to see the queue, half of the queue UI is not appearing for the one second I swipe up then it reappears after the animation has ended and the screen stays without input again.  It's like the queue was cut in half so we are able to see what's behind it (the currently selected tab in Innertune)

**Comment by Malopieds** [comment]
> I see gradient often loading a whole lot after the start of the song.. I don't know what's the cause of this issue...

It's the time for extracting the colors can't do much about it. But it takes max 2 seconds. 

There is already an issue for the second thing you described.

**Comment by Richard38907** [comment]
Z-Huang pushed the 0.5.7 update with a new animation when opening the mini player (dark-> black with a smooth gradient). And since this new animation, the opening animation is now lagging like this version of this fork😕.
https://github.com/user-attachments/assets/eb5610f3-1503-41d9-963a-9e83d99f8b69

Sadly the video doesn't render correctly the lag.

**Comment by Malopieds** [comment]
I didn't add this animation, but try to fix the old one. Could you tell me if it's better (given that this would be a debug version so, if you can build it it would be better)

**Comment by Richard38907** [comment]
I haven't built a debug apk. What do I need?

**Comment by Malopieds** [comment]
If you want to test it as it will be on release you need androidStudio or build it by CLI. There is the debug version, but it's slower than release version, or you can't wait for a next release.

**Comment by d4f5409d** [comment]
> And when I swipe up to see the queue, half of the queue UI is not appearing for the one second I swipe up then it reappears after the animation has ended and the screen stays without input again.  It's like the queue was cut in half so we are able to see what's behind it (the currently selected tab in Innertune)

![Screenshot_2024-08-23-11-17-28-96_56635af9b6ad4512d4187a5a49185a25.jpg](https://github.com/user-attachments/assets/68bbe0de-72ef-4d12-96fa-b3d30163f009)

**Comment by Malopieds** [comment]
> There is already an issue for the second thing you described.

I've already responded (and fixed) that. You're not using the latest debug version.

**Comment by Richard38907** [comment]
Re-Ui lagging on 0.5.9

🎉The bug is only present on OnePlus devices (with snapdragon Soc) I haven't tested on other OnePlus specs (Mediatek Soc...) yet. 

I tested now on a Google Pixel 7 and the UI is smooth like before.
If other users encounter the problem with other brands/phone specs, please leave a feedback here 🙂.

**Comment by ReallyYug** [comment]
> Re-Ui lagging on 0.5.9
> 
> 🎉The bug is only present on OnePlus devices (with snapdragon Soc) I haven't tested on other OnePlus specs (Mediatek Soc...) yet.
> 
> I tested now on a Google Pixel 7 and the UI is smooth like before. If other users encounter the problem with other brands/phone specs, please leave a feedback here 🙂.

@Malopieds the quoted text and https://github.com/z-huang/InnerTune/issues/1456 proves it ig and I have nord 300 if that helps

**Comment by Malopieds** [comment]
Yeah but there is no reason it bugs on some and other no. I don't know what I can do here :/

**Comment by ReallyYug** [comment]
Ye idk, well we just gotta deal with it ig

**Comment by Richard38907** [comment]
I cannot help you more. @YUG38 do you encounter the glitch too on your Nord N300?
![Screenshot_20240825-184708.png](https://github.com/user-attachments/assets/37f0f8e6-3551-4252-8417-73e44021af9e)

It's a Mediatek soc.

**Comment by ReallyYug** [comment]
@Richard38907 yes the app is "laggy" for me and idk what Mediatek soc is sorry

**Comment by Richard38907** [comment]
Mediatek is a brand of phone's processor. OnePlus uses Qualcomm or Mediatek socs on their devices.

So the bug seems to occur on all OnePlus devices(Mtk/Snapdragon) but I don't know why. I know that Samsung uses Snapdragon SOCs but I don't know the real reason of that bug😕.

**Comment by ReallyYug** [comment]
Oh ok I see, well we can't do anything ig, just gotta buy a new phone 😛 ( jk )

**Comment by Richard38907** [comment]
OnePlus is a good brand. I don't plan to change now😅.

**Comment by ReallyYug** [comment]
Did anyone check if the og repo app is lagging too or it's just this repo maybe that will make it a little easier to figure out what the issue is

**Comment by Richard38907** [comment]
Tested on Samsung Galaxy A40. Worked perfectly but after the update, the glitch occurs.

**Comment by Malopieds** [comment]
Is it better now?

**Comment by Richard38907** [comment]
Sadly no, the frame drops are still present on AMOLED mode.
